Rpackage/R/initial.pruning.R

```diff
@@ -0,0 +1,44 @@
+init_pruning <- structure(function(tree){
+  
+  ### id of inner node
+  ter_id <- nodeids(tree, terminal = TRUE)
+  all_id <- nodeids(tree, terminal = FALSE)
+  inner_id <- all_id[-ter_id]
+  
+  ### if there is no inner node
+  if(length(inner_id) == 0){
+    return(NULL)
+  }
+  
+  for(n in 1:length(inner_id)){
+    ### node id from the last
+    node_id <- tail(inner_id, n = n)[1]
+    ### partynode object
+    t <- nodeapply(tree, ids = node_id)[[1]]
+    node <- as.partynode(t)
+    
+    ### left right child ids and partynodes
+    left_kid_id <- sapply(kids_node(node), id_node)[1]
```

Comment:
Do `sapply(kids_node(node), id_node)` only once and use index

Reply:
Also, no need to get the ids and then cut the tree at the kids. kids_node already returns the kid nodes.